File: IntellijBuild/src/main/java/com/learning/project2/models/TestHistory.java
package com.learning.project2.models;


import com.learning.project2.models.test.Test;
import com.learning.project2.models.user.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.Instant;

@Table(name="p2_test_history")
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class TestHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "test_id", nullable = false)
    private Test test;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "date_taken")
    private Instant dateTaken;

}

File: AngularBuild/src/app/models/test_history.ts
export class TestHistory 
{
    id: number; 
    test_id: number; 
    user_id: number; 
    date_taken: Date = new Date(); 

    constructor(data: any)
    {
        data = data || {}; 
        this.id = data.id; 
        this.test_id = data.test_id; 
        this.user_id = data.user_id; 
    }

}